### PR TITLE
Hard: refactor borrow/borrows and deposit/deposits query

### PR DESCRIPTION
**Pull request by denalimarsh**
- [x] Merge borrow/borrows into query borrows
- [x] Merge deposit/deposits into query deposits
- [x] Clean up querier, update examples/descriptions, etc.

### CODE REVIEW

File: x/hard/client/cli/query.go [hunk 1]
```diff
@@ -272,10 +309,42 @@ func queryBorrowsCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
+					return fmt.Errorf("failed to unmarshal borrow balance: %w", err)
+				}
+
+				if len(denom) > 0 {
+					if balance.AmountOf(denom).Equal(sdk.ZeroInt()) {
+						return fmt.Errorf("user %s has no borrow balance for denom %s", owner, denom)
+					}
+				}
+
+				return cliCtx.PrintOutput(balance)
```
Comment (karzak) [inline x/hard/client/cli/query.go:338]: Same as deposits, return the object not the balance

**Review by karzak — CHANGES_REQUESTED** [review]

File: x/hard/client/cli/query.go [hunk 2]
```diff
@@ -145,10 +145,42 @@ func queryDepositsCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
+					return fmt.Errorf("failed to unmarshal deposit balance: %w", err)
+				}
+
+				if len(denom) > 0 {
+					if balance.AmountOf(denom).Equal(sdk.ZeroInt()) {
+						return fmt.Errorf("user %s has no deposit balance for denom %s", owner, denom)
+					}
+				}
+
+				return cliCtx.PrintOutput(balance)
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/hard/client/cli/query.go]: Fixed in https://github.com/Kava-Labs/kava/pull/761/commits/980e510bb3e4a15f48ec6fa48c9e8d6f2ab19192

File: x/hard/client/cli/query.go [hunk 3]
```diff
@@ -124,9 +121,12 @@ func queryDepositsCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
 		$ kvcli q hard deposits
-		$ kvcli q hard deposits --owner kava1l0xsq2z7gqd7yly0g40y5836g0appumark77ny --deposit-denom bnb
-		$ kvcli q hard deposits --deposit-denom ukava
-		$ kvcli q hard deposits --deposit-denom btcb`,
+		$ kvcli q hard deposits --owner kava1l0xsq2z7gqd7yly0g40y5836g0appumark77ny --denom bnb
+		$ kvcli q hard deposits --denom ukava
+		$ kvcli q hard deposits --denom btcb
+
+		Flag --owner will return the user's latest borrow balance including any outstanding interest.
+		Without flag --owner it is not guaranteed that returned user borrows include outstanding interest.`,
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/hard/client/cli/query.go]: Fixed in https://github.com/Kava-Labs/kava/pull/761/commits/e4bd838873c7cb543d33bd8cfa4dfb7f77e8e751

File: x/hard/client/cli/query.go [hunk 4]
```diff
@@ -252,8 +286,11 @@ func queryBorrowsCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
 
 		Example:
 		$ kvcli q hard borrows
-		$ kvcli q hard borrows --borrower kava1l0xsq2z7gqd7yly0g40y5836g0appumark77ny
-		$ kvcli q hard borrows --borrow-denom bnb`,
+		$ kvcli q hard borrows --owner kava1l0xsq2z7gqd7yly0g40y5836g0appumark77ny
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/hard/client/cli/query.go:258]: Updated in https://github.com/Kava-Labs/kava/pull/761/commits/a1b9845df11d58ac63214fe86404ccbdeb604ede

File: x/hard/client/cli/query.go [hunk 5]
```diff
@@ -39,14 +39,11 @@ func GetQueryCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
 		queryParamsCmd(queryRoute, cdc),
 		queryModAccountsCmd(queryRoute, cdc),
 		queryDepositsCmd(queryRoute, cdc),
-		queryDepositCmd(queryRoute, cdc),
 		queryClaimsCmd(queryRoute, cdc),
 		queryBorrowsCmd(queryRoute, cdc),
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/hard/client/cli/query.go:44]: Good catch. Added it to available query cmds and refactored it to accept `denom` param in https://github.com/Kava-Labs/kava/pull/761/commits/fbd421c949f243ee251f6ba4879350658b2b4096

File: x/hard/client/cli/query.go (hunk 1 above)
Comment (denalimarsh) [reply x/hard/client/cli/query.go]: Updated in https://github.com/Kava-Labs/kava/pull/761/commits/e4bd838873c7cb543d33bd8cfa4dfb7f77e8e751

**Review by karzak — APPROVED** [review]
tACK

**Pull request merged**